MSAL/src/public/configuration/MSALGlobalConfig.h

```diff
@@ -0,0 +1,72 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class MSALHTTPConfig;
+@class MSALTelemetryConfig;
+@class MSALLoggerConfig;
+@class MSALCacheConfig;
+@class MSALAuthority;
+
+@interface MSALGlobalConfig : NSObject
+
+/*! Network configuration, , refer to MSALHTTPConfig.h for more detail */
+@property (class, readonly) MSALHTTPConfig *httpConfig;
+/*! Telemetry configurations, refer to MSALTelemetryConfig.h for more detail */
+@property (class, readonly) MSALTelemetryConfig *telemetryConfig;
+/*! Logger configurations, refer to MSALLoggerConfig.h for more detail */
+@property (class, readonly) MSALLoggerConfig *loggerConfig;
+/*! Cache configurations, refer to MSALCacheConfig.h for more detail */
+@property (class, readonly) MSALCacheConfig *cacheConfig;
+
+/*! List of known authorities that application should trust. */
```

Comment:
Add additional comment explaining that application should never put an authority here that's dynamically resolved and can be malicious.

Reply:
/*! List of known authorities that application should trust.
    Note that authorities listed here will bypass authority validation logic.
    Thus, it is advised not putting in here dynamically resolving authorities here.
 */